nixos/modules/virtualisation/amazon-image.nix

```diff
@@ -25,6 +25,9 @@ in
       { assertion = cfg.hvm;
         message = "Paravirtualized EC2 instances are no longer supported.";
       }
+      { assertion = cfg.efi -> cfg.hvm;
```

Comment:
I don't know if this is a strict requirement, but it reduces the possible configurations to support.